5.0/en/0x51-V51-OAuth2.md

```diff
@@ -0,0 +1,99 @@
+| **51.2.5** | [ADDED] Verify that Clients are utilizing the parameters "scope" and "authorization_details" to determine the requested/related resources and/or actions the access token are restricted to. | ✓ | ✓ | ✓ |
+
+## V51.3 Resource Server
+
+| # | Description | L1 | L2 | L3 |
+| :---: | :--- | :---: | :---: | :---: |
+| **51.3.1** | [ADDED] Verify that Resource Servers are using mechanisms for sender-constraining access tokens to prevent token replay, such as Mutual TLS for OAuth 2.0 or OAuth Demonstration of Proof of Possession (DPoP).  | ✓ | ✓ | ✓ |
+| **51.3.2** | [ADDED] Verify that access tokens are restricted to certain Resource Servers (audience restriction), preferably to a single Resource Server. Every Resource Server is obliged to verify, for every request, whether the access token sent with that request was meant to be used for that particular Resource Server. If not, the Resource Server must refuse to serve the respective request. | ✓ | ✓ | ✓ |
+| **51.3.3** | [ADDED] Verify that access tokens are restricted to certain resources and actions on Resource Servers or resources. Every Resource Server is obliged to verify, for every request, whether the access token sent with that request was meant to be used for that particular action on the particular resource. If not, the Resource Server must refuse to serve the respective request. | ✓ | ✓ | ✓ |
+
+## V51.4 Resource Owner
+
+| # | Description | L1 | L2 | L3 |
+| :---: | :--- | :---: | :---: | :---: |
+| **51.4.1** | [ADDED] Verify that the Resource Owner password credentials grant is not used. This grant type insecurely exposes the credentials of the Resource Owner to the client, increasing the attack surface of the application.  | ✓ | ✓ | ✓ |
+
+## Mapping
+
+This mapping shows which OAuth 2.0 requirements belongs to which OAuth-specific subsection from the original requirements structure that was proposed here: https://github.com/OWASP/ASVS/blob/5375e96bc99c47fe7b3430407cd469d0f9896180/5.0/en/0x12-V3-Session-management.md#v38-oauth-20-protocol
+
+| Section | Requirements |
+| :--- | :--- |
+| OAuth 2.0 Essential Basics | 51.1.1, 51.1.2 |
+| PCKE | 51.1.3, 51.1.4, 51.1.5, 51.3.1 |
+| Token Replay | 51.2.1, 51.2.2 |
+| Access Restriction | 51.2.3, 51.2.4 |
+| Resource Owner | 51.2.5 |
+
+## OAuth 2.0 References
+
+* Original RFC: https://www.rfc-editor.org/info/rfc6750 & : https://www.rfc-editor.org/info/rfc6749
+* OAuth 2.0 Best Practices: https://datatracker.ietf.org/doc/html/draft-ietf-oauth-security-topics#name-best-practices
+* Mix-up attacks: https://datatracker.ietf.org/doc/html/draft-ietf-oauth-security-topics-18#mix_up
+* RFC9207 [I-D.ietf-oauth-iss-auth-resp]: https://datatracker.ietf.org/doc/html/draft-ietf-oauth-security-topics-18#section-2.1-4
+* & [I-D.ietf-oauth-iss-auth-resp]: https://datatracker.ietf.org/doc/html/draft-ietf-oauth-iss-auth-resp-00

```

Comment:
These two lines look a little confusing. What are the supposed to be pointing to? The same thing or different things? @csfreak92

Reply:
Sorry, which ones please @tghosth?

Reply:
> * RFC9207 [I-D.ietf-oauth-iss-auth-resp]: https://datatracker.ietf.org/doc/html/draft-ietf-oauth-security-topics-18#section-2.1-4
> * & [I-D.ietf-oauth-iss-auth-resp]: https://datatracker.ietf.org/doc/html/draft-ietf-oauth-iss-auth-resp-00

These two lines, I can't understand what they are supposed to be pointing to or telling me.

Reply:
They are pointing to different things. One is for OAuth 2.0 Security Best Current Practice and the other one is for OAuth 2.0 Authorization Server Issuer Identifier in Authorization Response. I think the confusion was with the titles in the [hard brackets here]. Let me fix that too.

Reply:
Fixed in #1880